string entry_name
---
string data_str